schemas/2015-05-21-preview/Microsoft.DevTestLabs.json

```diff
@@ -0,0 +1,69 @@
+{
+  "id": "http://schema.management.azure.com/schemas/2015-05-21-preview/Microsoft.DevTestLab.json#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Microsoft.DevTestLabs",
+  "description": "Azure DevTest Labs Resource Types",
+  "definitions": {
+    "labs": {
+      "description": "Required properties to create a DevTest Lab",
+      "type": "object",
+      "properties": {
+        "type": {
+          "enum": [
+            "Microsoft.DevTestLabs/labs"
+          ]
+        },
+        "apiVersion": {
+          "enum": [
+            "2015-05-21-preview"
+          ]
+        }
+      },
+      "required": [ "type", "apiVersion" ]
+    },
+    "environments": {
+      "description": "Required properties to create environment in an existing DevTest Lab",
+      "type": "object",
+      "properties": {
+        "type": {
+          "enum": [
+            "Microsoft.DevTestLabs/environments"
+          ]
+        },
+        "apiVersion": {
+          "enum": [
+            "2015-05-21-preview"
+          ]
+        },
+        "properties": {
+          "type": "object",
+          "properties": {
+            "labId": { "type": "string" },
+            "vms": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "required": [ "name", "vmTemplateName", "size", "userName", "isAuthenticationWithSshKey" ],
+                "oneOf": [
```

Comment:
just to confirm, one and only one of the two properties should be used in the same time.